### ISSUE TITLE
Captured webpage pagemarks behave oddly

### ISSUE THREAD

**Issue by sopoforic** [burtonator/polar-bookshelf#1461]
In beta 16, on chrome 84, pagemarks in captured web documents behave in an unsettling fashion.

Testing with [this wikipedia page](https://en.wikipedia.org/wiki/Sorcerous_Stabber_Orphen).

To begin, I make a pagemark just below the infobox. This seems to work as expected:

![chrome_r0pXekefPl](https://user-images.githubusercontent.com/78205/91612107-b1ea5e80-e94a-11ea-894e-ecaf3ffcf7df.png)

However, attempting to drag the pagemark down below the first line results in it snapping back to its original position. It is necessary to drag it down until it is overlapping the second line before it will stay in place:

![chrome_NW6fmDAu85](https://user-images.githubusercontent.com/78205/91612125-c0387a80-e94a-11ea-99d3-dee57ca5ac0c.png)

This could be mitigated by having the pagemark snap to acceptable positions when dragging, rather than moving smoothly and then snapping back when let go.

Additionally, if I drag the pagemark and then resize the browser window, the pagemark snaps back to its former location. Only changes in position caused by right-clicking and selecting "Create Pagemark to Point" survive resizing. However, dragging the pagemark, then closing the tab and reopening it *does* result in the pagemark staying where I put it, so the change is being persisted properly.

**Comment by burtonator** [comment]
> This could be mitigated by having the pagemark snap to acceptable positions when dragging, rather than moving smoothly and then snapping back when let go.

Yes... I think this is possible but I need to spend more time on on it ... I might have to completely rework the way we do resizing for that to work.  

I agree that it's a big jarring though.

**Comment by sopoforic** [comment]
This is an even bigger problem for me, now that epubs are generally working. Dragging the bottom of pagemarks often causes them to snap to the bottom of the page, and otherwise they usually snap back a large distance. Right-clicking and setting a pagemark usually puts it where I want, so the problem shouldn't be fundamental to the pagemark system.

Besides actually marking how far I've read, I like (in PDFs where it works, at least) to use pagemarks to help me as I read, by dragging the pagemark down as I go down the page. This makes it easier to read documents with long lines without getting lost and re-reading a line. It's impossible to do this with epubs.
